### ISSUE TITLE
Wait for custom event to start building

### ISSUE THREAD

**Issue by viki53** [mean-expert-official/loopback-sdk-builder#246]
#### What type of issue are you creating?
- [ ] Bug
- [x] Enhancement
- [ ] Question

#### What version of this module are you using?
- [ ] 2.0.10 (Stable)
- [ ] 2.1.0-rc.n (2.1 Release Candidate n)
- [x] Other

Write other if any:

2.1.0-rc.2

#### Please add a description for your issue:

Hi,

We (@ebarault and myself) are using a mixin to extend our models. this mixin can only start on the `booted` event, which means that when the SDK builder start, its job isn't done and our models don't have all the proper methods exposed in the generated SDK.

Is there any way to provide a custom event name to the SDK (maybe via an environment variable) to wait a little longer before building?

**Comment by jonathan-casarrubias** [comment]
Hi @viki53 thanks for reaching me out.

Hey so, the SDK Builder already awaits for the booted event, I believe it is registered before of the mixing one and it is executed before.... Maybe we need to migrate it to 'started'

please modify the ./node_modules/.bin/lb-sdk

remove the following lines

````js
84 //assertLoopBackVersion();
85 if (app.booting) {
86  app.on('booted', verifyPath);
87 } else {
88  verifyPath();
89 }
````

change all of them to

````js
app.on('started', verifyPath);
````

and see if now works

**Comment by ebarault** [comment]
Hi @jonathan-casarrubias, I just did, on "started": it launches the server and does not give the hand back to the sdk builder.

**Comment by jonathan-casarrubias** [comment]
I see, what is happening is that within the server.js file it does not start the app when it is exported.

The problem is that we now would need to start the app from the SDK Builder, but that would be done before the boot....

Maybe for your specific case this could help.

server/server.js

````js
// Bootstrap the application, configure models, datasources and middleware.
// Sub-apps like REST API are mounted via boot scripts.
boot(app, __dirname, function(err) {
  if (err) throw err;

  // start the server if `$ node server.js`
  if (require.main === module) // Try removing this condition
    app.start();
});
````

**Comment by ebarault** [comment]
inded @jonathan-casarrubias : it works. Not totally clean though, as it still starts the server which is confusing for the user. I'd maybe handle this with an env var to explicitly not launch the server when building the sdk.

Do you think it's safe to remove the 'if (require.main === module)' permanently ?


---
see : despite the server is launched I can still click Y to proceed.
````
WARNING: The targeted SDK Directory is not empty, do you want to erase its content? (Y/n) 

12/12/2016 20:46:01.841 - info: Web server (development) listening at: http://0.0.0.0:3000/api/v1
12/12/2016 20:46:01.842 - info: Swagger API explorer is available at http://0.0.0.0:3000/explorer

(clicked Y)

Loading Framework angular2
Generating "sdk" for the API endpoint "/api/v1"
...
````

**Comment by jonathan-casarrubias** [comment]
Hey @ebarault I'm glad it worked,

It is safe to permanently remove that condition if you don't need to start the server from outside of loopback, in most of the cases you won't do that. If server.js is your main node.js module then there should be no issue.

The server is always booted regardless of what you have done, is just that is now serving the booted objects into the port 3000, which does not make that much of difference. Anyway I think it is complex to ask developers to follow this procedure everytime if they don't have a mixing of this type.

So, I will keep this as an approach for people that requires this, but I will leave the current booting process, since this is a very unusual case.

Cheers
Jon
